cmd/update.go

```diff
@@ -34,9 +34,18 @@ under $GOPATH/bin and automatically updates commands to the latest version.`,
 	cmd.Flags().BoolP("dry-run", "n", false, "perform the trial update with no changes")
 	cmd.Flags().BoolP("notify", "N", false, "enable desktop notifications")
 	cmd.Flags().StringSliceP("exclude", "e", []string{}, "specify binaries which should not be updated (delimiter: ',')")
+	if err := cmd.RegisterFlagCompletionFunc("exclude", cobra.NoFileCompletions); err != nil {
+		panic(err)
+	}
 	cmd.Flags().StringSliceP("main", "m", []string{}, "specify binaries which update by @main or @master (delimiter: ',')")
+	if err := cmd.RegisterFlagCompletionFunc("main", cobra.NoFileCompletions); err != nil {
+		panic(err)
+	}
 	// cmd.Flags().BoolP("main-all", "M", false, "update all binaries by @main or @master (delimiter: ',')")
 	cmd.Flags().IntP("jobs", "j", runtime.NumCPU(), "Specify the number of CPU cores to use")
+	if err := cmd.RegisterFlagCompletionFunc("jobs", completeNCPUs); err != nil {
+		panic(err)
```

Comment:
@scop 
"gup" is not a library, it's a small-scale code. I don't think "gup" encounters situations where panic() is necessary. It would be preferable to output error messages instead.

Reply:
I wrote a long comment, but the conclusion is "no need for modifications (this code is ok, good job)"

>私の意見では、gup がライブラリだったら、パニックは全く受け入れられないでしょう :)

This time, the part where you added panic() is within the new method. If gup were a library, I would tolerate embedding panic() in such places. I use panic() as an assert. That is, when something that I believe should never happen occurs, it indicates (panic()) that the programmer is misusing the method.

However, in the case of a command-line interface, users may not necessarily be developers, so I want to display more understandable error messages. Furthermore, putting panic() makes it harder to test, so I don't like using panic().

However, writing unit tests for initialization code is absurd. Therefore, I would like to adopt your opinion. Reading your thoughtful opinion, I felt that you are an excellent engineer.